code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -13,481 +13,374 @@ What are the archived variables for?
+//inverse
 /datum/gas_mixture/proc/get_true_breath_pressure(partial_pressure)
-	return (partial_pressure * BREATH_VOLUME) / (R_IDEAL_GAS_EQUATION * temperature)
+	return (partial_pressure * BREATH_VOLUME) / (R_IDEAL_GAS_EQUATION * return_temperature())
 
-///Mathematical proofs:
-/**
+//Mathematical proofs:
+/*
 get_breath_partial_pressure(gas_pp) --> gas_pp/total_moles()*breath_pp = pp
 get_true_breath_pressure(pp) --> gas_pp = pp/breath_pp*total_moles()
-
 10/20*5 = 2.5
 10 = 2.5/5*20
-**/
+*/
+
+/datum/gas_mixture/turf
+
+/*
+/mob/verb/profile_atmos()
+	/world{loop_checks = 0;}
+	var/datum/gas_mixture/A = new
+	var/datum/gas_mixture/B = new
+	A.parse_gas_string("o2=200;n2=800;TEMP=50")
+	B.parse_gas_string("co2=500;plasma=500;TEMP=5000")
+	var/pa
+	var/pb
+	pa = world.tick_usage
+	for(var/I in 1 to 100000)
+		B.transfer_to(A, 1)
+		A.transfer_to(B, 1)
+	pb = world.tick_usage
+	var/total_time = (pb-pa) * world.tick_lag
+	to_chat(src, "Total time (gas transfer): [total_time]ms")
+	to_chat(src, "Operations per second: [100000 / (total_time/1000)]")
+	pa = world.tick_usage
+	for(var/I in 1 to 100000)
+		B.total_moles();
+	pb = world.tick_usage
+	total_time = (pb-pa) * world.tick_lag
+	to_chat(src, "Total time (total_moles): [total_time]ms")
+	to_chat(src, "Operations per second: [100000 / (total_time/1000)]")
+	pa = world.tick_usage
+	for(var/I in 1 to 100000)
+		new /datum/gas_mixture
+	pb = world.tick_usage
+	total_time = (pb-pa) * world.tick_lag
+	to_chat(src, "Total time (new gas mixture): [total_time]ms")
+	to_chat(src, "Operations per second: [100000 / (total_time/1000)]")
+*/
+
+//these probably should be migrated to c++ eventually for m a x i m u m s p e e e e d
+
+///Distributes the contents of two mixes equally between themselves
+///Returns: bool indicating whether gases moved between the two mixes
+/datum/gas_mixture/proc/equalize(datum/gas_mixture/other)
+	. = FALSE
+	if(abs(return_temperature() - other.return_temperature()) > MINIMUM_TEMPERATURE_DELTA_TO_SUSPEND)
+		. = TRUE
+		var/self_heat_cap = heat_capacity()
+		var/other_heat_cap = other.heat_capacity()
+		var/new_temp = (return_temperature() * self_heat_cap + other.return_temperature() * other_heat_cap) / (self_heat_cap + other_heat_cap)
+		set_temperature(new_temp)
+		other.set_temperature(new_temp)
+
+	var/min_p_delta = 0.1
+	var/total_volume = return_volume() + other.return_volume()
+	var/list/gas_list = get_gases() | other.get_gases()
+	for(var/gas_id in gas_list)
+		//math is under the assumption temperatures are equal
+		if(abs(get_moles(gas_id) / return_volume() - other.get_moles(gas_id) / other.return_volume()) > min_p_delta / (R_IDEAL_GAS_EQUATION * return_temperature()))
+			. = TRUE
+			var/total_moles = get_moles(gas_id) + other.get_moles(gas_id)
+			set_moles(gas_id, total_moles * (return_volume() / total_volume))
+			set_moles(gas_id, total_moles * (other.return_volume() / total_volume))
```

Comment:
This looks new, why are we using pressure here?

Reply:
It's not actually new it's just a helper proc we had that cit did not. It probably should be migrated to c++ but it's not super critical and I want everything working correctly first.